library/Fission/IPFS/Peer.hs

```diff
@@ -54,5 +62,45 @@ connect peer@(Peer peerID) = IPFSProc.run ["swarm", "connect"] (UTF8.textToLazyB
   (ExitFailure _ , _, _) -> Left $ CannotConnect peer
   (ExitSuccess   , _, _) -> Right ()
 
+peerAddressRe :: Regex
+peerAddressRe = mkRegex "^/ip[46]/([a-zA-Z0-9.:]*)/"
+
+extractIPfromPeerAddress :: String -> Maybe String
+extractIPfromPeerAddress peer = peer
+  & matchRegex peerAddressRe
+  <&> Data.List.head
+
+isExternalIPv4 :: Text -> Bool
+isExternalIPv4 ip = ip
+  & T.unpack
+  & extractIPfromPeerAddress
+  & maybe "" T.pack
+  & Net.IPv4.decode
+  <&> Net.IPv4.reserved
+  & maybe False not
+
+
+filterExternalPeers :: [Peer] -> [Peer]
+filterExternalPeers = Data.List.filter (isExternalIPv4 . peer)
+
+getExternalAddress :: (MonadIO m
+  , MonadReader cfg m
+  , HasProcessContext cfg
+  , HasLogFunc cfg
+  , Has IPFS.BinPath cfg
+  , Has IPFS.Timeout cfg)
+  => m (Either Error [Peer])
```

Comment:
💅 

The linter will probably complain (and if it doesn't we should configure it to).

Either:

```haskell
getExternalAddress :: ( MonadRIO          cfg m
                      , HasProcessContext cfg
                      , HasLogFunc        cfg
                      , Has IPFS.BinPath  cfg
                      , Has IPFS.Timeout  cfg
                      ) 
                   => m (Either Error [Peer])

-- Or --

getExternalAddress :: MonadRIO          cfg m
                   => HasProcessContext cfg
                   => HasLogFunc        cfg
                   => Has IPFS.BinPath  cfg
                   => Has IPFS.Timeout  cfg
                   => m (Either Error [Peer])
```

I tend to use the second one because it's compact, but the first does make a clearer visual distinction between constraints and types.

Reply:
for right now the first one to me is both what comes out of my type signature generator and what is more readable (because of the visual difference between `=>` and `->`).

However I like how the second version looks. I would say lets keep the current convention :)